src/lib/certificate-manager.ts

```diff
@@ -148,11 +152,65 @@ export async function openCertificate() {
 	shell.showItemInFolder( CA_CERT_PATH );
 }
 
+/**
+ * Checks if the root CA certificate is already trusted by the system
+ * @returns A promise that resolves to true if the certificate is trusted, false otherwise
+ */
+export async function isRootCATrusted(): Promise< boolean > {
+	if ( ! fs.existsSync( CA_CERT_PATH ) ) {
+		return false;
+	}
+
+	if ( process.platform === 'win32' ) {
+		try {
+			// Execute certutil with more specific validation
+			const { stdout } = await execFilePromise( 'certutil', [
+				'-verify',
+				'-urlfetch',
+				CA_CERT_PATH,
+			] );
+
+			const hasValidPolicies =
+				stdout.includes( 'Verified Application Policies:' ) &&
+				stdout.includes( 'Server Authentication' );
+
+			// Only consider the certificate trusted if it has the Server Authentication policy.
+			return hasValidPolicies;
+		} catch ( error ) {
+			return false;
+		}
+	} else if ( process.platform === 'darwin' ) {
+		try {
+			await execFilePromise( 'security', [
+				'verify-cert',
+				'-c',
```

Comment:
```suggestion
				'-r',
```
The `-r` option is used to pass a path to a root cert file. I think its main feature is that it allows us to specify _both_ `-r` and `-c`. We don't do that here, but I still see no reason not to use `-r`.

Reply:
It makes to check for the root certificate explicitly.